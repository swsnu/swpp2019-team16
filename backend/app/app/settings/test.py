from .base import *
import environ
from redis import ConnectionPool

env = environ.Env()
env.read_env('env/.env.development')

DEBUG = True

# Database
# https://docs.djangoproject.com/en/2.1/ref/settings/#databases

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.sqlite3',
        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
    }
}

REDIS_CONNECTION_POOL = ConnectionPool(host=env('REDIS_HOST'), port=6379, db=0)
