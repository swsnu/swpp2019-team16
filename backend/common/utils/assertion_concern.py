class AssertionConcern:

    def assert_not_none(self, val, msg):
        if val is None:
            raise TypeError(msg)
