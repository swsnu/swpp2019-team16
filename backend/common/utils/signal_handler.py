import asyncio
import functools
import signal


def register_signal_handler(loop, shutdown):
    for signame in ('SIGINT', 'SIGTERM'):
        loop.add_signal_handler(
            getattr(signal, signame),
            functools.partial(asyncio.ensure_future,
                              shutdown(signame, loop)))


async def shutdown_process(self, sig, loop):
    print('caught {0}'.format(sig))
    tasks = [task for task in asyncio.Task.all_tasks() if task is not
             asyncio.tasks.Task.current_task()]

    list(map(lambda task: task.cancel(), tasks))
    results = await asyncio.gather(*tasks, return_exceptions=True)

    print('finished awaiting cancelled tasks, results: {0}'
          .format(results))
    loop.stop()
