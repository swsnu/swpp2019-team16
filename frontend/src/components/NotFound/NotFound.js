import React, { useCallback } from 'react';
import { withRouter } from 'react-router-dom';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/styles';
import theme from '../../lib/styles/theme';

const useStyles = makeStyles({
  root: {
    position: 'absolute',
    left: 0,
    top: 0,
    bottom: 0,
    right: 0,
    backgroundColor: theme.palette.primary.main,
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  container: {
    marginTop: '25px',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  whiteBox: {
    boxShadow: '0 0 8px rgba(0, 0, 0, 0.25)',
    padding: '2rem',
    width: '1000px',
    background: 'white',
    borderRadius: '2px',
    justifyContent: 'center',
    alignItems: 'center',
  },
  button: {
    background: theme.palette.primary.dark,
    borderRadius: 5,
    color: 'white',
    cursor: 'pointer',
    paddingLeft: '1rem',
    paddingRight: '1rem',
    paddingBottom: '0.3rem',
    paddingTop: '0.3rem',
    marginBottom: '10px',
    marginTop: '10px',
    marginLeft: '10px',
    marginRight: '10px',
  },
});

function NotFound({ history }) {
  const styles = useStyles();

  const onClickGoHome = useCallback(() => {
    history.push('/');
  }, [history]);

  return (
    <div className={styles.root}>
      <div className={styles.whiteBox}>
        <div className={styles.container}>
          <Typography variant="h1">404</Typography>
        </div>
        <div className={styles.container}>
          <Typography variant="h1">Page Not Found</Typography>
        </div>
        <div className={styles.container}>
          <Typography variant="h2">
            This is not the page you are looking for
          </Typography>
        </div>
        <div className={styles.container}>
          <div
            className={styles.button}
            children="Go Home"
            variant="contained"
            fullwidth="false"
            onClick={onClickGoHome}
          />
        </div>
      </div>
    </div>
  );
}
export default withRouter(NotFound);
