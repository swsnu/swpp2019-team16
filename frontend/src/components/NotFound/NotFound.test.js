import React from 'react';
import { renderWithRedux } from 'test/utils';
import { MemoryRouter, Route } from 'react-router-dom';
import NotFound from './NotFound';

describe('<NotFound />', () => {
  it('SHOULD match with snapshot', async () => {
    const { container } = renderWithRedux(
      <MemoryRouter initialEntries={['']}>
        <Route component={NotFound} />
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
