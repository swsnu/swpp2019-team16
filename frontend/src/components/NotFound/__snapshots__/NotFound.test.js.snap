// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> SHOULD match with snapshot 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="makeStyles-whiteBox-3"
    >
      <div
        class="makeStyles-container-2"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1"
        >
          404
        </h1>
      </div>
      <div
        class="makeStyles-container-2"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1"
        >
          Page Not Found
        </h1>
      </div>
      <div
        class="makeStyles-container-2"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2"
        >
          This is not the page you are looking for
        </h2>
      </div>
      <div
        class="makeStyles-container-2"
      >
        <div
          class="makeStyles-button-4"
          fullwidth="false"
          variant="contained"
        >
          Go Home
        </div>
      </div>
    </div>
  </div>
</div>
`;
